internal/check/check.go

```diff
@@ -0,0 +1,153 @@
+package check
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"strconv"
+	"sync"
+	"text/tabwriter"
+	"time"
+
+	"golang.org/x/sync/semaphore"
+)
+
+type RunFunc func(ctx context.Context) (string, error)
+
+type Check struct {
+	Name        string
+	Description string
+
+	Run RunFunc
+
+	LastRun           time.Time
+	CachedCheckErr    error
+	CachedCheckOutput string
+}
+
+func (c *Check) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		Name              string    `json:"name"`
+		Description       string    `json:"description"`
+		LastRun           time.Time `json:"last_run"`
+		CachedCheckErr    string    `json:"cached_check_err"`
+		CachedCheckOutput string    `json:"cached_check_output"`
+	}{
+		Name:              c.Name,
+		Description:       c.Description,
+		LastRun:           c.LastRun,
+		CachedCheckErr:    errString(c.CachedCheckErr),
+		CachedCheckOutput: c.CachedCheckOutput,
+	})
+}
+
+type HealthChecker struct {
+	mu     sync.Mutex
+	Checks []Check
+}
+
+func (c *HealthChecker) MarshalJSON() ([]byte, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	return json.Marshal(c.Checks)
+}
+
+func (c *HealthChecker) Run() {
+	for {
+		time.Sleep(60 * time.Second)
```

Comment:
Depending on the check - I think we'll need to have them specify when they'd like to run. In general, a ping check will run on a difference cadence (every few minutes) as opposed to a disk speed check (once on startup)? Because of this, I think it makes sense to:

-  have each `check` spin up a goroutine that handles calling the check on a certain cadence. We should have only a few checks per service < 20 - so this check should be cheap. 
- The overall healthChecker will just scrape the output instead of invoking the checker directly